anyblok/column.py

```diff
@@ -1524,3 +1556,21 @@ def update_table_args(self, registry, Model):
         key.update(str(enum).encode('utf-8'))
         name = self.fieldname + '_' + key.hexdigest() + '_types'
         return [CheckConstraint(constraint, name=name)]
+
+
+@CompareType.add_comparator(String, String)
+@CompareType.add_comparator(String, Selection)
+@CompareType.add_comparator(String, Sequence)
+def compare_strings(col1, col2):
+    if col1.size != col2.size:
+        raise FieldException(
+            "You can't add a foreign key on %r(%d) -> %r(%d)" % (
+                col1.__class__, col1.size, col2.__class__, col2.size))
+
+
+@CompareType.add_comparator(String, Color)
+def compare_string_to_color(col1, col2):
+    if col1.size != col2.max_length:
+        raise FieldException(
+            "You can't add a foreign key on %r(%d) -> %r(%d)" % (
+                col1.__class__, col1.size, col2.__class__, col2.max_length))
```

Comment:
Not sure it's a good idea but we probably could be factorize with `compare_strings` method likes::

```python
size1 = col1.getattr("size", col1.getattr("max_length"))
```
otherwise use previous comment for raising nice message

Reply:
col1 is a String and col2 a Color, i  dont think is a good way to estimate the good attribute